modules/peridynamics/src/actions/MechanicsActionPD.C

```diff
@@ -0,0 +1,220 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Factory.h"
+#include "FEProblemBase.h"
+#include "Conversion.h"
+#include "MooseObjectAction.h"
+#include "MechanicsActionPD.h"
+
+#include "libmesh/string_to_enum.h"
+
+registerMooseAction("PeridynamicsApp", MechanicsActionPD, "add_aux_variable");
+registerMooseAction("PeridynamicsApp", MechanicsActionPD, "add_kernel");
+registerMooseAction("PeridynamicsApp", MechanicsActionPD, "add_user_object");
+registerMooseAction("PeridynamicsApp", MechanicsActionPD, "setup_quadrature");
+
+template <>
+InputParameters
+validParams<MechanicsActionPD>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Class for setting up peridynamic kernels");
+  params.addRequiredParam<std::vector<VariableName>>(
+      "displacements", "Nonlinear variable names for the displacements");
+  MooseEnum formulation_option("Bond OrdinaryState NonOrdinaryState", "Bond");
+  params.addParam<MooseEnum>("formulation",
+                             formulation_option,
+                             "Available peridynamic formulation options: " +
+                                 formulation_option.getRawNames());
+  MooseEnum stabilization_option("Force Self", "Self");
+  params.addParam<MooseEnum>(
+      "stabilization",
+      stabilization_option,
+      "Available stabilization options for Non-Ordinary State Based Peridynamics: " +
+          stabilization_option.getRawNames());
+  params.addParam<bool>(
+      "full_jacobian",
+      false,
+      "Parameter to set whether to use full jacobian for state based formulation or not");
+  params.addParam<bool>("finite_strain_formulation",
+                        false,
+                        "Parameter to set whether the formulation is finite strain or not");
+  params.addParam<VariableName>("temperature", "Nonlinear variable name for the temperature");
+  params.addParam<VariableName>("out_of_plane_strain",
+                                "Nonlinear variable name for the out_of_plane strain for "
+                                "plane stress using SNOSPD formulation");
+  params.addParam<bool>(
```

Comment:
This should just be determined by whether you are asking for small or finite strain.